### ISSUE TITLE
Custom collectionType binding with fluent-builder

### ISSUE THREAD

**Issue by mastersang** [mklemm/jaxb2-rich-contract-plugin#63]
In my binding file I apply a custom List for a specific list element via 
<jaxb:property collectionType="MyList" />

When I use fluent-builder, the generarated builder code always uses ArrayList instead. I've read the source code of the plugin and it seems you're hard-coding it to use ArrayList (PluginContext.arrayListClass).

Is there anyway to change this behaviour? I could work on the source code but I would need some directions.

**Comment by mklemm** [comment]
You're right, the ArrayList type has been hardcoded. It would certainly be possible to apply collection types. I'm looking into it...

**Comment by LiviaH** [comment]
Have the same issue, seems the fluent builder didnt check a field is null or not.
If the collectionType is set each field is initialized with the collection type instance. But the fluent builder replace the collection instead of adding the values. Can be its better to check the field is null or not.
I have an problem using default values. Default values  overwritten with null. 
To reproduce this behavior just create and field with default value
<element name="defValue" type="string" default="myValue" minOccurs="1" maxOccurs="1" />
Class generation will create a field and initialize it
    @XmlElement(required = true, defaultValue = "myValue")
    protected String defValue = "myValue";
But the init Method overwrite it with null:
 _product.defValue = this.defValue;
At this line this.defValue is null because withDefValue is never called for default values.

Can be an null check can solve both. But I'm not sure, is there any other possibility to use default values together with the builder api?

**Comment by mklemm** [comment]
Planned for 4.2.0

**Comment by mklemm** [comment]
Re default values: This is a different issue. Rudimentary handling of attribute default values in fluent-builder will also be added in 4.2.0
